src/Application/ServerApplication.php

```diff
@@ -107,10 +107,15 @@ public function addClient(Protocol\LivereloadProtocol $client)
 
     public function removeClient(Protocol\LivereloadProtocol $client)
     {
+        $this->clients = array_values(array_filter($this->clients, function($client) {
+            return !empty($client->app) && !empty($client->conn);
+        }));
+        
         $index = array_search($client, $this->clients, true);
         if($index == false){
```

Comment:
If $client is the first element of $this->clients (means $this->clients[0]), the result of $index is 0.
Using strong type compare ($index === false) is better.